File: solutions/1827/main.py
from typing import * 

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        res = 0 
        for i in range(1, len(nums)): 
            new_val = max(nums[i-1]+1, nums[i])
            res += new_val - nums[i]
            nums[i] = new_val 
        return res 

File: solutions/1827/main.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        let mut dp = 0; 
        let mut pre = nums[0]; 
        for i in 1..nums.len() {
            let cur = (pre+1).max(nums[i]); 
            dp += cur - nums[i]; 
            pre = cur; 
        }
        dp 
    }
}